src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/playback/GerritMissedEventsPlaybackManager.java

```diff
@@ -0,0 +1,438 @@
+            for (ChangeBasedEvent evt: events) {
+                logger.info("({}) Processing missed event {}", serverName, evt);
+                if (receivedEventCache.contains(evt)) {
+                    logger.info("({}) Event already triggered...skipping trigger.", serverName);
+                } else {
+
+                    //do we have this event in the time slice?
+                    EventTimeSlice slice = serverTimestamps.get(serverName);
+                    long currentEventCreatedTime = ((GerritTriggeredEvent)evt).getEventCreatedOn().getTime();
+                    if (slice.getTimeSlice() == currentEventCreatedTime) {
+                        if (slice.getEvents().contains(evt)) {
+                            logger.info("({}) Event already triggered from time slice...skipping trigger.", serverName);
+                            continue;
+                        }
+                    }
+                    logger.info("({}) Triggering: {}", serverName, evt);
+                    PluginImpl.getInstance().getServer(serverName).triggerEvent(evt);
+                    receivedEventCache.add((ChangeBasedEvent)evt);
+                    logger.debug("Added event {} to received cache for server: {}", evt, serverName);
+                    persist(evt);
+                }
+            }
+        } catch (UnsupportedEncodingException e) {
+            logger.error("Error building URL for playback query: " + e.getMessage(), e);
+        } catch (IOException e) {
+            logger.error("Error accessing URL for playback query: " + e.getMessage(), e);
+        }
+        playBackComplete = true;
+        logger.info("Processing completed for server: {}", serverName);
+    }
+
+    /**
+     * Log when the connection goes down.
+     */
+    @Override
+    public void connectionDown() {
+        logger.info("connectionDown for server: {}", serverName);
+    }
+
+    /**
+     * This allows us to persist a last known alive time
+     * for the server.
+     * @param event Gerrit Event
+     */
+    @Override
+    public void gerritEvent(GerritEvent event) {
+        if (!isSupported()) {
+            return;
+        }
+
+        if (event instanceof GerritTriggeredEvent) {
+            logger.info("Recording timestamp due to an event {} for server: {}", event, serverName);
+            GerritTriggeredEvent triggeredEvent = (GerritTriggeredEvent)event;
+            persist(triggeredEvent);
+            //add to cache
+            if (!playBackComplete) {
+                if (!receivedEventCache.contains(event)) {
+                    receivedEventCache.add((ChangeBasedEvent)event);
+                    logger.debug("Added event {} to received cache for server: {}", event, serverName);
+                } else {
+                    logger.debug("Event {} ALREADY in received cache for server: {}", event, serverName);
+                }
+            } else {
+                receivedEventCache = new Vector<ChangeBasedEvent>();
```

Comment:
This will create a new object for every event coming in during normal operations, it seems a bit unnecessary exercising of the heap.